low-level/sspl_ll_d

```diff
@@ -544,42 +566,80 @@ def _sensors_actuators_factory(sspl_threaded_modules, msgQlist, operating_system
     return (sspl_threaded_modules, msgQlist)
 
 
-def _run_thread_capture_errors(curr_module, sspl_threaded_modules, msgQlist, conf_reader, product, resume):
-    """Run the given thread and log any errors that happen on it.
-    Will stop all sspl_threaded_modules if one of them fails."""
-    try:
-        # Suspend module threads
-        if resume == False:
-            curr_module.suspend()
-        # Each module is passed a reference list to message queues so it can transmit
-        # internal messages to other modules as desired
-        logger.info("Starting: %s" % curr_module.name())
-        curr_module.start_thread(conf_reader, msgQlist, product)
-
-    except ThreadException as ex:
-        #TODO: Restart thread instead of shutdown
-        logger.error("%s, error is unrecoverable , shutting down %s" %
-            (ex, curr_module.name()))
-        curr_module.shutdown()
-    except Exception as ex:
-        # Populate an actuator response message and transmit back to HAlon that we have a fatal error
-        # error_msg = "SSPL-LL encountered an error, terminating service Error, restarting daemon:{}" \
-        #             .format(str(logger.exception(ex)))
-        # jsonMsg   = ThreadControllerMsg(curr_module.name(), error_msg).getJson()
-        # curr_module._write_internal_msgQ(EgressProcessor.name(), jsonMsg)
+def _run_thread_capture_errors(module, msgQlist, conf_reader, product, resume,
+                               recovery_count=0, recovery_interval=0):
+    logger.info("Starting: %s" % module.name())
 
-        # Shut down any running threads, error is non-recoverable
-        # shutdown_handler()
+    # Initialize persistence cache for sensor status
+    node_id = Conf.get(GLOBAL_CONF, NODE_ID_KEY, "SN01")
+    per_data_path = os.path.join(MODULE_CACHE_DIR,
+                f'{module.name().upper()}_{node_id}')
```

Comment:
prepend node-id, to identify sensor easily

Reply:
Need to discuss on this cache part.

If sensor failure is captured, here fault alert will be raised and cached.
I couldn't figure out where to put state change('fault' to 'fault_resolved') !!
At this time I removed persistent cache.